app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,634 @@
+            height: 38,
+        },
+        loadingContainer: {
+            alignItems: 'center',
+            backgroundColor: theme.centerChannelBg,
+            height: 70,
+            justifyContent: 'center',
+        },
+        loadingText: {
+            color: changeOpacity(theme.centerChannelColor, 0.6),
+        },
+        noResultContainer: {
+            flexGrow: 1,
+            flexDirection: 'row',
+            alignItems: 'center',
+            justifyContent: 'center',
+        },
+        noResultText: {
+            fontSize: 26,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+        },
+        separator: {
+            height: 1,
+            flex: 0,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+        },
+    };
+});
+
+function IntegrationSelector(
+    {dataSource, data, isMultiselect, selected, handleSelect,
+        currentTeamId, componentId, getDynamicOptions, options, teammateNameDisplay}: Props) {
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const style = getStyleSheet(theme);
+    const searchBarInput = {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.2),
+        color: theme.centerChannelColor,
+        fontSize: 15,
+    };
+    const intl = useIntl();
+
+    // HOOKS
+    const [integrationData, setIntegrationData] = useState<DataType>(data || []);
+    const [loading, setLoading] = useState<boolean>(false);
+    const [term, setTerm] = useState<string>('');
+    const [searchResults, setSearchResults] = useState<DataType>([]);
+    const [multiselectSelected, setMultiselectSelected] = useState<MultiselectSelectedMap>({});
+    const [currentPage, setCurrentPage] = useState<number>(INITIAL_PAGE);
+    const [next, setNext] = useState<boolean>(VALID_DATASOURCES.includes(dataSource));
+    const [customListData, setCustomListData] = useState<DataType | UserProfileSection[]>([]);
+
+    // Callbacks
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults([]);
+    }, []);
+
+    // This is the button to submit multiselect options
+    const rightButton = useMemo(() => {
+        const base = buildNavigationButton(
+            SUBMIT_BUTTON_ID,
+            'integration_selector.multiselect.submit.button',
+            undefined,
+            intl.formatMessage({id: 'integration_selector.multiselect.submit', defaultMessage: 'Done'}),
+        );
+        base.enabled = true;
+        base.showAsAction = 'always';
+        base.color = theme.sidebarHeaderTextColor;
+        return base;
+    }, [theme.sidebarHeaderTextColor, intl]);
+
+    const handleSelectItem = useCallback((item: Selection) => {
+        if (!isMultiselect) {
+            handleSelect(item);
+            close();
+            return;
+        }
+
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS: {
```

Comment:
The only difference between the cases is what is the key to use (the id or the value). This would be way more legible if you use the case to set the `key`, and do the rest of the operations in outside of the switch.

Reply:
the type casting is different, I could try to do it in a cleaner way.

Reply:
Done!